compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3693,6 +3693,10 @@ private boolean isAssignableToUnionType(BType source, BType target, Set<TypePair
         return true;
     }
 
+    private boolean isJsonAnydataOrUserDefinedUnion(BType type) {
```

Comment:
Shall we pass type tag here?

Reply:
I think it's better to pass the type since it's less work on the caller side. Will extract type.tag out to a variable though.